test/realm_test.dart

```diff
@@ -268,14 +268,26 @@ Future<void> main([List<String>? args]) async {
     expect(realm.write(() => realm.add(carTwo, update: true)), carTwo);
   });
 
-  test('Realm write after realm is closed', () {
+  test('Realm write after realm is closed', () async {
     var config = Configuration.local([Car.schema]);
     var realm = getRealm(config);
 
     final car = Car('Tesla');
 
     realm.close();
-    expect(() => realm.write(() {}), throws<RealmClosedError>("Cannot access realm that has been closed"));
+    _writeAndExpectException<RealmClosedError>(realm, "Cannot access realm that has been closed");
```

Comment:
We should not be changing all the test to use this API. Lets use this method only where it was needed for the double write.

Reply:
The reason to use it was because I had to test all the cases where async write throws exceptions. And async write when realm is closed is one of them.

Reply:
I see you point. In this case could we rename this to `_expectAllWritesToThrow<T extends Exception>`
For the `RealmClosedError` I think we should leave it in the handle getter but also check in each write method if the realm is closed and throw a `RealmException("Write failed. The realm is already closed.")`

Reply:
If you search by `RealmClosedError` you can see that it is used in the the handle getters for list, results, realm, realm object, session and so many places and so many tests. I suggest if you want we to do this it to be in a separate PR. It will introduce bricking changes.

Reply:
I can rename these methods to `_expectWriteToThrow `and `_expectWriteAsyncToThrow` if you prefer. I need to have them separately because for the one of the tests 'Realm write inside another write throws' I expect only `write` and `beginWrite` to throw inside another write. The async writes are allowed inside another write block.

Reply:
`_expectAllWritesToThrow` and `_expectAllAsyncWritesToThrow`?

Reply:
Lets leave `RealmCloseError` as it is

Reply:
Done